tempesta_fw/sock_srv.c

```diff
@@ -483,7 +502,7 @@ tfw_sock_srv_connect_srv(TfwServer *srv)
 	 * that parallel execution can't happen with the same socket.
 	 */
 	list_for_each_entry(srv_conn, &srv->conn_list, list) {
-		tfw_server_get(srv);
+		tfw_sock_srv_conn_activate(srv, srv_conn);
 		tfw_sock_srv_connect_try_later(srv_conn);
```

Comment:
Seems like `tfw_sock_srv_connect_try_later()` is always called after `tfw_sock_srv_conn_activate()`. Let's move it there.

Reply:
There is also `tfw_sock_srv_connect_try_later()` call here https://github.com/tempesta-tech/tempesta/blob/f4ef380a113a4d194a13ab3ae5bb4f719dac9797/tempesta_fw/sock_srv.c#L309, and it seems that we needn't increase server reference count in this path.